core/src/main/java/org/frankframework/pipes/RegExPipe.java

```diff
@@ -0,0 +1,195 @@
+		UNIX_LINES(Pattern.UNIX_LINES);
+
+		private final @Getter int flag;
+
+		RegExFlag(int flag) {
+			this.flag = flag;
+		}
+	}
+
+	private String regex = "";
+	private String flags = "";
+
+	private Pattern pattern;
+
+	private PipeForward thenForward;
+	private PipeForward elseForward;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+
+		final int flags = this.flags.isEmpty() ? 0 : Arrays.stream(this.flags.split(","))
+				.map(flag -> RegExFlag.valueOf(flag).getFlag())
+				.reduce(0, (a, b) -> a | b);
+
+		if (this.regex == null || this.regex.isEmpty()) {
+			throw new ConfigurationException("Value for `regex is mandatory.`");
+		}
+
+		this.pattern = Pattern.compile(this.regex, flags);
+
+		this.thenForward = findForward(THEN_FORWARD);
+		this.elseForward = findForward(ELSE_FORWARD);
+
+		if (this.thenForward == null) {
+			throw new ConfigurationException("has no forward with name [" + THEN_FORWARD + "]");
+		}
+
+		if (this.elseForward == null) {
+			throw new ConfigurationException("has no forward with name [" + ELSE_FORWARD + "]");
+		}
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
+		String sInput;
+		if (Message.isEmpty(message)) {
+			sInput = "";
+		} else {
+			try {
+				sInput = message.asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, "cannot open stream", e);
+			}
+			if (sInput == null) {
+				throw new PipeRunException(this, "Input message is empty");
+			}
+		}
+
+		Matcher matcher = this.pattern.matcher(sInput);
+		XmlBuilder matches = new XmlBuilder("matches");
+
+		int index = 1;
+		boolean foundMatch = false;
+		while (matcher.find()) {
+			foundMatch = true;
+			matches.addSubElement(matchToXml(matcher, index));
+			index += 1;
+		}
+
+		if (!foundMatch) {
+			return new PipeRunResult(elseForward, null);
+		}
+
+		Message result = matches.asMessage();
+		return new PipeRunResult(thenForward, result);
+	}
+
+	private XmlBuilder matchToXml(Matcher matcher, int index) {
+		int groups = matcher.groupCount();
+
+		XmlBuilder match = new XmlBuilder("match");
+		match.addAttribute("index", index);
+		match.addAttribute("value", matcher.group());
+
+		for (int i = 1; i <= groups; i++) {
+			XmlBuilder group = new XmlBuilder("group");
+			group.addAttribute("index", i);
+			group.setValue(matcher.group(i));
+
+			match.addSubElement(group);
+		}
+
+		return match;
+	}
+
+	/**
+	 * The regex expression to match against the input.
+	 */
+	@Mandatory
+	public void setRegex(String regex) {
+		this.regex = regex;
+	}
+
+	/**
+	 * Comma seperated list of flags, which changes the behavior of the regex expression.
+	 *
+	 * Possible values: CANON_EQ, CASE_INSENSITIVE, COMMENTS, DOT_ALL, LITERAL, MULTILINE, UNICODE_CASE, UNICODE_CHARACTER_CLASS and UNIX_LINES.
+	 *
+	 * Example: MULTILINE,DOT_ALL,CASE_INSENSITIVE
+	 */
+	public void setFlags(String flags) {
```

Comment:
I dislike having the developer enter a comma seperated list of enum values. Is there are better way to input a list?

Reply:
An idea I had is to allow `N` parameters with the name `flag`, but that also doesn't improve typesafety, so I guess that that isn't much better.

Reply:
Zie https://github.com/frankframework/frankframework/blob/master/core/src/main/java/org/frankframework/http/rest/ApiListener.java#L281 -- ik denk dat dat ook voor jou gebruikt kan worden.